src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,80 @@
-import React from 'react';
+import React, { useEffect, useState, MouseEvent } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { actions as curentTodoActions } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+  const [userName, setUserName] = useState<string>('');
+  const curentTodo: Todo | null = useAppSelector(state => state.currentTodo);
 
-      <Loader />
+  const dispatch = useAppDispatch();
+  const removeTodo = () => (dispatch(curentTodoActions.removeTodo()));
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  const removeHandler = (event: MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    removeTodo();
+  };
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        if (curentTodo) {
+          const user = await getUser(curentTodo.userId);
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">fugiat veniam minus</p>
+          setUserName(user.name);
+        }
+      } catch (error) {
+        throw new Error();
+      }
+    };
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+    fetchUser();
+  }, []);
```

Comment:
Add 'curentTodo' as a dependency to useEffect to refetch user data when the curentTodo changes.